jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/traveler/TravelerSerializer.java

```diff
@@ -19,42 +23,61 @@
 public class TravelerSerializer {
 
     public TravelerInformation travelerInfo;
+    public final int ADVISORY = 0;
+    public final int WORKZONE = 1;
+    public final int GENERICSIGN = 2;
+    public final int SPEEDLIMIT = 3;
+    public final int EXITSERVICE = 4;
+    int contentType;
     public TravelerSerializer(String jsonInfo){
         PERUnalignedCoder coder = J2735.getPERUnalignedCoder();
-//      MsgCount msgCnt = new MsgCount(1);
-//        MinuteOfTheYear moty = new MinuteOfTheYear(1);
-//
-//        int var = 1;
-//        byte[] myvar = var.getBytes();
-//        UniqueMSGID msdId = new UniqueMSGID(myvar);
-//
-//        URL_Base url = new URL_Base("www.google.con");
-//
-////        TravelerDataFrame(SSPindex sspTimRights,
-////                TravelerInfoType frameType, MsgId msgId, DYear startYear,
-////                MinuteOfTheYear startTime, MinutesDuration duratonTime,
-////                SignPrority priority, SSPindex sspLocationRights,
-////                Regions regions, SSPindex sspMsgRights1,
-////                SSPindex sspMsgRights2, Content content, URL_Short url)
-////        {
-//
-//        TravelerInformation decoder = new TravelerInformation();
-//        TravelerInformation travelerContent = new TravelerInformation();
-//
-//        TravelerDataFrame dataFrame = new TravelerDataFrame();
-//
-//        TravelerDataFrameList dataFrameList = new TravelerDataFrameList();
-//        TravelerInformation.Regional regional = new TravelerInformation.Regional();
-//
-//
-//
-//        AbstractData abstractTim = new TravelerInformation(msgCnt, moty, msdId, url, dataFrameList, regional);
-//
-//        System.out.print(coder.encode(abstractTim));
 
+        travelerInfo = new TravelerInformation();
+
+        //Get fully populated TIMcontent string
         JSONObject obj = new JSONObject(jsonInfo);
+
+        int frameList = obj.getJSONObject("timContent").getJSONArray("travelerDataFrame").length(); //Check the dataframe count
+
+        //Populate pojo's for TIM
         String msgcnt = obj.getJSONObject("timContent").getString("msgcnt");
+        validateMessageCount(msgcnt);
+        travelerInfo.setMsgCnt(new MsgCount(Integer.parseInt(msgcnt)));
+
+        //TODO the travelerDataFrame can be a list
+        TravelerDataFrameList dataFrames = new TravelerDataFrameList();
+        for (int z = 1; z <= frameList; z++)
+        {
+            TravelerDataFrame dataFrame = new TravelerDataFrame();
+            String curFrame = "df" + z;
+            //Populate pojo's for part1-header
+            TravelerDataFrame part1 = buildTravelerMessagePart1(dataFrame, obj.getJSONObject("timContent").getJSONArray("travelerDataFrame").getJSONObject(0));
```

Comment:
highly inefficient to call obj.getJSONObject() method for the same key multiple times. This is not a simple getter? Every time you get the value of a key, it actually searches for it from the beginning of the object. Save the JSONObject if  you need to reuse it. Doing so will also eliminate a ton of duplicate hard-coded strings in the code too which is something sonarqube will complain about for sure.